pkg/component/component.go

```diff
@@ -125,6 +125,140 @@ func ApplyConfig(client kclient.ClientInterface, envSpecificInfo envinfo.EnvSpec
 	})
 }
 
+// Contains checks to see if the component exists in an array or not
+// by checking the name, type and namespace.
+// We ignore all other aspects (labels, annotations, etc.) as those are
+// propagated by the Kubernetes cluster when retrieving the component and we're only
+// interested if there is a duplicate entry.
+func Contains(component Component, components []Component) bool {
+	for _, comp := range components {
+		if (component.Name == comp.Name) && (component.Spec.Type == comp.Spec.Type) && (component.Namespace == comp.Namespace) {
+			return true
+		}
+	}
+	return false
+}
+
+// RemoveDuplicateComponentsForListingOutput is for listing purposes, we remove all duplicate components which have the same
+// kubernetes instance LABEL as each other.
+func RemoveDuplicateComponentsForListingOutput(components []Component) []Component {
```

Comment:
Could you please write some unit tests for this function, to have a visual of the cases it is covering?

Reply:
Done!